### ISSUE TITLE
How does xUnit compare/contrast with C++ Catch

### ISSUE THREAD

**Issue by mwpowellhtx** [xunit/xunit#1549]
I need to know, how it compares with something like *C++ Catch*?

Chiefly, in *Catch*, or similar *BDD* frameworks, **Test Cases** are organized into different **Sections**. Things in scope for a parent *Section* live through the next *Section*. *Sections* can be **nested**, and each *Section **leaf*** is guaranteed one execution path through the *terminal* leafs.

Generally, from an *NUnit* or *xUnit* perspective, I know how to organize *OO* unit tests generally using minimalist *OO* principles. That's no problem. Beyond that, how are the individual tests executed? ***Facts*** and/or ***Theories***?

In other words, what sort of context/state can I expect when execution lands in a given test method?

Thanks!

**Comment by mwpowellhtx** [comment]
Actually, I may have just answered my own question with this:

```C#
protected delegate void BehaviorDrivenDelegate(string title, Action action);
```

```C#
protected readonly BehaviorDrivenDelegate Given;
protected readonly BehaviorDrivenDelegate Section;
```

```C#
protected TestFixtureBase(ITestOutputHelper @out)
{
  Out = @out;

  Given = Section = (title, action) =>
  {
    Out.WriteLine(title);

    action();
  };
}
```

It's not quite the same, in that there are call stacks involved, and state preservation going on, but it's close enough.

Thoughts?

**Comment by bradwilson** [comment]
Have you considered using a BDD-style framework like [xBehave.net](http://xbehave.github.io/)?

**Comment by mwpowellhtx** [comment]
I'll look into that, thanks.
